File: v2/mopidyAgent/js/mopidyAgentRabbitListener.js
var Listener = function(amqp, worker) {
    this.amqp = amqp;
    this.worker = worker;

    this.amqp.connect('amqp://localhost', (err, conn) => {
        conn.createChannel((err, ch) => {
            var ex = 'jukeboxExchange';
            ch.assertExchange(ex, 'topic', {durable: false});

            ch.assertQueue('', {exclusive: true}, (err, q) => {
                ch.bindQueue(q.queue, ex, 'album');
                ch.consume(q.queue, (msg) => {
                    console.log("MopidyAgentRabbitListener %s:'%s'", msg.fields.routingKey, msg.content.toString());
                    this.worker.play(msg.content.toString());
                }, {noAck: true});
            });

            ch.assertQueue('', {exclusive: true}, (err, q) => {
                ch.bindQueue(q.queue, ex, 'command.mopidy');
                ch.consume(q.queue, (msg) => {
                    console.log("MopidyAgentRabbitListener %s:'%s'", msg.fields.routingKey, msg.content.toString());
                    this.worker.handleCommand(msg.content.toString());
                }, {noAck: true});
            });
        });
    });
}

module.exports = Listener;

File: v2/mopidyAgent/mopidyAgentRabbitListener.py
#!/usr/bin/env python
import pika
from mopidyAgentWorker import MopidyAgentWorker

class MopidyAgentRabbitListener:
    def __init__(self):
        ex = 'jukeboxExchange'
        self.mopidyAgent = MopidyAgentWorker()

        connection = pika.BlockingConnection(pika.ConnectionParameters(host='localhost'))
        channel = connection.channel()
        channel.exchange_declare(exchange=ex, type='topic')

        result = channel.queue_declare(exclusive=True)
        queue_name = result.method.queue
        channel.queue_bind(exchange=ex, queue=queue_name, routing_key='album')

        print(' [*] Waiting for logs. To exit press CTRL+C')

        channel.basic_consume(self.callback, queue=queue_name, no_ack=True)
        channel.start_consuming()

    def callback(self, ch, method, properties, body):
        print("ALBUMRETRIEVER MOPIDY AGENT: Should now call mopidyAgent worker")
        print(" [x] %r:%r" % (method.routing_key, body))
        self.mopidyAgent.doWork(body)

if __name__ == '__main__':
    print("Running mopidyAgent rabbitlistener standalone")
    marl = MopidyAgentRabbitListener()
